src/ContractCallDataEncoder.cpp

```diff
@@ -382,6 +389,8 @@ QVariant ContractCallDataEncoder::decodeRawArray(SolidityType const& _type, byte
 		value.populate(&rawParam);
```

Comment:
I've noticed that. The test above (line 375) is here to prevent to continue if "pos" is really important. 
What I saw is that when the memory is in an intermediate state "pos" could be really huge, this test was enough to prevent a crash in that case. 
but I guess I should be more precise you're right.